behaviour/graql/language/insert.feature

```diff
@@ -200,64 +295,1159 @@ Feature: Graql Insert Query
+      | PER | key   | ref:0 |
+      | EMP | key   | ref:1 |
+    Then uniquely identify answer concepts
+      | p   | r   |
+      | PER | EMP |
+
+
+  Scenario: when inserting a ternary relation that both owns an attribute and has an attribute as a roleplayer, both attributes are created
+    Given graql define
+      """
+      define
+      residence sub relation,
+        relates resident,
+        relates place-of-residence,
+        has is-permanent,
+        key ref;
+      person plays resident;
+      address sub attribute, value string, plays place-of-residence;
+      is-permanent sub attribute, value boolean;
+      """
+    Given the integrity is validated
+    When graql insert
+      """
+      insert
+      $p isa person, has name "Homer", has ref 0;
+      $perm true isa is-permanent;
+      $r (resident: $p, place-of-residence: $addr) isa residence, has is-permanent $perm, has ref 0;
+      $addr "742 Evergreen Terrace" isa address;
+      """
+    When the integrity is validated
+    Then get answers of graql query
+      """
+      match $r (place-of-residence: $addr) isa residence, has is-permanent $perm; get;
+      """
+    Then concept identifiers are
+      |     | check | value                         |
+      | RES | key   | ref:0                         |
+      | ADD | value | address:742 Evergreen Terrace |
+      | PER | value | is-permanent:true             |
+    Then uniquely identify answer concepts
+      | r   | addr | perm |
+      | RES | ADD  | PER  |
+
+
+  Scenario: when inserting a relation with multiple role players, they are all added
+    When graql insert
+      """
+      insert
+      $p1 isa person, has name "Gordon", has ref 0;
+      $p2 isa person, has name "Helen", has ref 1;
+      $c isa company, has name "Morrisons", has ref 2;
+      $r (employer: $c, employee: $p1, employee: $p2) isa employment, has ref 3;
+      """
+    When the integrity is validated
+    When get answers of graql query
+      """
+      match
+        $r (employer: $c) isa employment;
+        $c has name $cname;
+      get $cname;
+      """
+    Then concept identifiers are
+      |     | check | value          |
+      | MOR | value | name:Morrisons |
+    Then uniquely identify answer concepts
+      | cname |
+      | MOR   |
+    When get answers of graql query
+      """
+      match
+        $r (employee: $p) isa employment;
+        $p has name $pname;
+      get $pname;
+      """
+    Then concept identifiers are
+      |     | check | value       |
+      | GOR | value | name:Gordon |
+      | HEL | value | name:Helen  |
+
+
+  Scenario: an additional role player can be inserted into a relation
+    Given graql insert
+      """
+      insert $p isa person, has ref 0; $r (employee: $p) isa employment, has ref 1;
+      """
+    Given the integrity is validated
+    When graql insert
+      """
+      match $r isa employment; insert $r (employer: $c) isa employment; $c isa company, has ref 2;
```

Comment:
split over two lines - easier to read if match/insert and separated

Reply:
(and elsewhere too)

Reply:
Done